src/network_filter_list.rs

```diff
@@ -1,37 +1,94 @@
-use std::{collections::HashMap, collections::HashSet, sync::Arc};
+use std::{collections::HashMap, collections::HashSet, fmt};
 
 use serde::{Deserialize, Serialize};
 
-use crate::filters::network::NetworkFilter;
-use crate::filters::network::NetworkMatchable;
+use crate::filters::fb_network::flat::fb;
+use crate::filters::fb_network::{FlatNetworkFilter, FlatNetworkFiltersListBuilder};
+use crate::filters::network::{
+    NetworkFilter, NetworkFilterMask, NetworkFilterMaskHelper, NetworkMatchable,
+};
 use crate::optimizer;
 use crate::regex_manager::RegexManager;
 use crate::request::Request;
 use crate::utils::{fast_hash, Hash};
 
-#[derive(Serialize, Deserialize, Default)]
-pub(crate) struct NetworkFilterList {
-    #[serde(serialize_with = "crate::data_format::utils::stabilize_hashmap_serialization")]
-    pub(crate) filter_map: HashMap<Hash, Vec<Arc<NetworkFilter>>>,
+pub struct CheckResult {
+    pub filter_mask: NetworkFilterMask,
+    pub modifier_option: Option<String>,
+    pub raw_line: Option<String>,
+}
+
+impl From<&NetworkFilter> for CheckResult {
+    fn from(filter: &NetworkFilter) -> Self {
+        Self {
+            filter_mask: filter.mask,
+            modifier_option: filter.modifier_option.clone(),
+            raw_line: filter.raw_line.clone().map(|v| *v),
+        }
+    }
+}
+
+impl fmt::Display for CheckResult {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        if let Some(ref raw_line) = self.raw_line {
+            write!(f, "{}", raw_line)
+        } else {
+            write!(f, "{}", self.filter_mask)
+        }
+    }
+}
+
+impl NetworkFilterMaskHelper for CheckResult {
+    #[inline]
+    fn has_flag(&self, v: NetworkFilterMask) -> bool {
+        self.filter_mask.contains(v)
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct NetworkFilterList {
+    pub(crate) flatbuffer_memory: Vec<u8>,
+    pub(crate) filter_map: HashMap<Hash, Vec<u32>>,
+    pub(crate) unique_domains_hashes_map: HashMap<Hash, u16>,
+}
+
+impl Default for NetworkFilterList {
+    fn default() -> Self {
+        Self {
+            flatbuffer_memory: Default::default(),
+            filter_map: Default::default(),
+            unique_domains_hashes_map: Default::default(),
+        }
+    }
```

Comment:
`#[derive(Serialize, Deserialize, Default)]` should automatically produce this `Default` implementation already

Reply:
Oh,,god, I forgot to add `Default` it the derive list... I'll fix it in one of the follow ups